pkg/security/admission/admission.go

```diff
@@ -310,25 +278,50 @@ func (c *constraint) getNamespace(name string, ns *kapi.Namespace) (*kapi.Namesp
 	return c.client.Core().Namespaces().Get(name)
 }
 
-// getMatchingSecurityContextConstraints returns constraints from the store that match the group,
-// uid, or user of the service account.
-func getMatchingSecurityContextConstraints(store cache.Store, userInfo user.Info) ([]*kapi.SecurityContextConstraints, error) {
-	matchedConstraints := make([]*kapi.SecurityContextConstraints, 0)
+// SetInformers implements WantsInformers interface for constraint.
+func (c *constraint) SetInformers(informers shared.InformerFactory) {
+	c.sccLister = informers.SecurityContextConstraints().Lister()
+}
 
-	for _, c := range store.List() {
-		constraint, ok := c.(*kapi.SecurityContextConstraints)
-		if !ok {
-			return nil, errors.NewInternalError(fmt.Errorf("error converting object from store to a security context constraint: %v", c))
-		}
+// Validate defines actions to vallidate security admission
+func (c *constraint) Validate() error {
+	if c.sccLister == nil {
+		return fmt.Errorf("SecurityContextConstraint plugin not valid")
+	}
+	return nil
+}
+
+// SCCMatcher defines interface for SecurityContextConstraint matcher
+type SCCMatcher interface {
```

Comment:
Can you move this interface and impl to `pkg/security/scc` (new pacakge) so that your future `pkg/security/registry/pspsubjectreview` doesn't have to depend on an admission package?